lib/compat/wordpress-6.3/script-loader.php

```diff
@@ -42,18 +42,39 @@ function _gutenberg_get_iframed_editor_assets() {
 	$wp_styles->registered  = $current_wp_styles->registered;
 	$wp_scripts->registered = $current_wp_scripts->registered;
 
-	wp_enqueue_style( 'wp-block-editor-content' );
-	// To do: investigate why this is not enqueued through enqueue_block_assets,
-	// as styles for non-core blocks are.
-	wp_enqueue_style( 'wp-block-library' );
+	// We do not need reset styles for the iframed editor.
+	$wp_styles->done = array( 'wp-reset-editor-styles' );
+
 	wp_enqueue_script( 'wp-polyfill' );
+	// Enqueue the `editorStyle` handles for all core block, and dependencies.
+	wp_enqueue_style( 'wp-edit-blocks' );
+
+	if ( 'site-editor.php' === $pagenow ) {
+		wp_enqueue_style( 'wp-edit-site' );
+	}
+
+	if ( current_theme_supports( 'wp-block-styles' ) ) {
+		wp_enqueue_style( 'wp-block-library-theme' );
+	}
```

Comment:
I discovered that both `wp-edit-site` and `wp-block-library-theme` were also not correctly enqueued (compat warning).

Reply:
@ellatrix I see `wp-block-library-theme` is already a dependency of `wp-edit-blocks` in both [core](https://github.com/WordPress/wordpress-develop/blob/44038873e242ae10ea2c20187a431feaae166ac8/src/wp-includes/script-loader.php#L1651) and [gutenberg](https://github.com/WordPress/gutenberg/blob/85704f4bd051908bdd8c0ff66306d7831d6c60c9/lib/client-assets.php#L359). Why would we need to enqueue it explicitely?

Reply:
Actually. I don't see `wp-block-library-theme` enqueued anywhere (post or site editor).

Reply:
Upon the `init` action, `gutenberg_reregister_core_block_types` is called, which then calls `gutenberg_register_core_block_assets`, which dequeues `wp-block-library-theme` [under some scenarios](https://github.com/WordPress/gutenberg/blob/1868ba7c092c2ffe31fceb78a83f686dcde0230a/lib/blocks.php#L243). A note for later to investigate. I need to look to another thing first.

Reply:
I don't see it listed there as a dependency:

https://github.com/WordPress/gutenberg/blob/85704f4bd051908bdd8c0ff66306d7831d6c60c9/lib/client-assets.php#L315-L325

Reply:
I tested a bit how `wp-block-library-theme` works and this is what I see:

1. If the theme doesn't declare theme support for `wp-block-styles` it is not loaded. Can be tested using TwentyTwentyThree.
2. If the theme declares support for `wp-block-styles` but doesn't load editor styles via `add_editor_styles` is loaded as a dependency of `wp-edit-blocks`. I tested this by using TwentyTwentyTwo and [removing support for editor styles](https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-content/themes/twentytwentytwo/functions.php#L28).
3. If the theme declares support for `wp-block-styles` and loads editor styles via `add_editor_styles` it is loaded, just not as a dependency of `wp-edit-blocks`. Can be tested using TwentyTwentyTwo.

As core currently stands, this PR works as expected. Just checking for `wp-block-styles` supports is fine, and the `wp-block-library-theme` stylesheet loads for the scenarios 2 and 3 but not for 1. We can move forward with this PR.

---

A question I have that is separate from this PR is: why is there a need for separate scenarios 2 and 3? Isn't it the same as loading `wp-block-library-theme` stylesheet when the theme declares support for `wp-block-styles`? Pinging @aristath @gziolo and @ntsekouras as people that I see in the git blames for that area, in case they have thoughts.